src/lib/operators/projection.cpp

```diff
@@ -28,7 +28,23 @@ Projection::Projection(const std::shared_ptr<const AbstractOperator>& input_oper
                        const std::vector<std::shared_ptr<AbstractExpression>>& init_expressions)
     : AbstractReadOnlyOperator(OperatorType::Projection, input_operator, nullptr,
                                std::make_unique<OperatorPerformanceData<OperatorSteps>>()),
-      expressions(init_expressions) {}
+      expressions(init_expressions) {
+  /**
+   * Register as a consumer for all subplans of uncorrelated subqueries
```

Comment:
Same discussion as in the previous PR. For me, "subplan" is something different. Why not simply "for all uncorrelated subqueries"? Also, move the comment about correlated subqueries up here?